cpp/src/slice2swift/SwiftUtil.cpp

```diff
@@ -866,56 +614,64 @@ SwiftGenerator::writeServantDocSummary(IceInternal::Output& out, const Interface
         for (OperationList::const_iterator q = ops.begin(); q != ops.end(); ++q)
         {
             OperationPtr op = *q;
-            DocElements opdoc = parseComment(op);
+            CommentPtr opdoc = op->parseComment(swiftLinkFormatter, true, true);
             out << nl << "///";
-            out << nl << "///  - " << fixIdent(op->name()) << ": ";
-            if (!opdoc.overview.empty())
+            out << nl << "///  - " << fixIdent(op->name());
+            if (opdoc)
             {
-                writeDocSentence(out, opdoc.overview);
+                out << ": ";
+                StringList opdocOverview = opdoc->overview();
+                if (!opdocOverview.empty())
+                {
+                    writeDocSentence(out, opdocOverview);
+                }
             }
         }
     }
 
-    if (!doc.misc.empty())
+    StringList docMisc = doc->misc();
+    if (!docMisc.empty())
     {
-        writeDocLines(out, doc.misc, false);
+        writeDocLines(out, docMisc, false);
     }
 }
 
 void
 SwiftGenerator::writeMemberDoc(IceInternal::Output& out, const DataMemberPtr& p)
 {
-    DocElements doc = parseComment(p);
-
-    //
-    // Skip if there are no doc comments.
-    //
-    if (doc.overview.empty() && doc.misc.empty() && doc.seeAlso.empty() && doc.deprecateReason.empty() &&
-        !doc.deprecated)
+    CommentPtr doc = p->parseComment(swiftLinkFormatter, true, true);
+    if (!doc)
     {
         return;
     }
 
-    if (doc.overview.empty())
+    // Skip if there are no doc comments.
+    StringList docOverview = doc->overview();
+    StringList docMisc = doc->misc();
+    StringList docSeeAlso = doc->seeAlso(); // TODO we check seeAlso, but don't write any of in the generated comment?
```

Comment:
There are definitely still bugs in the `slice2swift` doc generation, I `TODO`ed the ones I found, and plan to fix them when I fix #2274.